cmd/crates/soroban-rpc/src/lib.rs

```diff
@@ -1010,11 +1006,9 @@ soroban config identity fund {address} --helper-url <url>"#
             xdr::ScVal::ContractInstance(xdr::ScContractInstance {
                 executable: xdr::ContractExecutable::Wasm(hash),
                 ..
-            }) => Ok(
-                contract::Spec::new(&self.get_remote_wasm_from_hash(hash).await?)
-                    .map_err(Error::CouldNotParseContractSpec)?
-                    .spec,
-            ),
+            }) => Ok(soroban_spec::read::from_wasm(
+                &self.get_remote_wasm_from_hash(hash).await?,
+            )?),
```

Comment:
This is the main part of the change. The `soroban-spec` crate already provides the same functionality that the `soroban-spec-tools` crate wraps and adds additional CLI related functionality to. There's no need for the RPC to use the `soroban-spec-tools` crate for this as it doesn't need the CLI related functionality.

Reply:
Nice !!